### DISCUSSION TITLE
Day(s) worth of firestore data going missing?

### DISCUSSION THREAD

**Discussion by TMcLoone** [invertase/react-native-firebase#6244] [Q&A]
Not sure if this is the right place to discuss this but wondering if anyone has experienced an issue where days worth of firestore data has gone missing for a user? My best guess it that is getting cached and never actually ends up getting sent to firestore then when the cache gets too big it just clears it. 
Seems to be randomly affecting iOS and Android users using 14.8.0. I haven't been able to recreate the issue but I've had multiple users report this.

**Comment by mikehardy** [comment]
Oh no! This may or may not be related to this: #6158 - for which I finally gave up on getting a fully-working firebase-android-sdk (newer versions of auth broke auth emulator, but fixed firestore) and issued [14.9.3 here yesterday](https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md#1493-2022-05-10) with a workaround that keeps the older firebase-android-sdk bom version but updates *just* firestore to get the fix

Does that seem like it might be your use case? You might ingest 14.9.3 as a hotfix and roll that out if possible, if so

That issue was probably present as early as 14.3.2 here *if* that's the issue though, as it was the last time we officially ingested a newer version of the underlying android sdk. ios has had a couple bumps in between.

That date range is pretty broad so normally it might make this hypothesis - the firestore reconnection issue - a little suspect, but your symptoms seem to really fit it. I would definitely update either way, since it could be it

**Reply by mikehardy** [reply]
Could simply be that in normal circumstances users are not able to sync cache for longer than expected vs expected data size, at which point you might investigate https://rnfirebase.io/reference/firestore/settings#cacheSizeBytes in combination with some sort of feature that would be project-specific, where you kept track of pending writes to see if it was really a problem normally - like https://stackoverflow.com/a/67631873/9910298 - I suppose you could write them to user prefs or async storage or something - basically somewhere that is *not* firestore since it is having the problem - to have a little trail, maybe daily?, of how many pending writes there were, and then when device had connectivity you could push that to backend and get some metrics / information on the problem.

Just brainstorming really, from the perspective of "If I saw this in my app, how would I even begin to solve it..."

**Reply by TMcLoone** [reply]
I saw that and wondered if that was the case but it's happening to iOS users too. Do you think it could still be that? Working on rolling out 14.9.3 today

**Reply by mikehardy** [reply]
That (symptoms on ios) bothers me as well, I'm not sure if ios sdk had same/similar issues as we had no problems ingesting those new SDK versions as soon as they came out, so we did and released them. It may be there was a connection restoration issue there too, I just don't know

But it led me down the path of thinking "okay, assume the new version does nothing, then what?" which was my second brainstorming comment. Tough problem! I'm not sure how much more help I can/will be but empathize and I'll certainly be listening if you learn anything

**Reply by TMcLoone** [reply]
> That (symptoms on ios) bothers me as well, I'm not sure if ios sdk had same/similar issues as we had no problems ingesting those new SDK versions as soon as they came out, so we did and released them. It may be there was a connection restoration issue there too, I just don't know
> 
> But it led me down the path of thinking "okay, assume the new version does nothing, then what?" which was my second brainstorming comment. Tough problem! I'm not sure how much more help I can/will be but empathize and I'll certainly be listening if you learn anything

ok thanks Mike! I'll let you know how things go.
